src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs

```diff
@@ -207,5 +185,10 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
                     removedCallback: delegate { graphicsResult.Dispose(); });
             }
         }
+
+        protected override void RemoveAdornmentFromAdornmentLayer_CallOnlyOnUIThread(SnapshotSpan span)
+        {
+            AdornmentLayer.RemoveAdornmentsByVisualSpan(new SnapshotSpan(span.Start.GetContainingLine().Start, span.End.GetContainingLine().End));
```

Comment:
Does this still have issues with empty lines, since the span will be empty in that case? A good example might be the "extra blank lines" analyzer we have?

Reply:
Ah the source of my confusion was the underlying code looked like it was doing OverlapsWith check on the span (which would always be false for a zero-width span), but no, it's actually got custom logic. @dpugh recently renamed it to avoid confusing people.